### PR TITLE
Fix: XCTL options

### PR DESCRIPTION
**Pull request by sb-broadcom**
## How Has This Been Tested?

Unit tests to check valid and invalid inputs

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+/** Checks CICS XCTL rules for required and invalid options */
+public class CICSXctlOptionsUtility extends CICSOptionsCheckBaseUtility {
+
+  public static final int RULE_INDEX = RULE_cics_xctl;
+
+  private static final Map<Integer, ErrorSeverity> DUPLICATE_CHECK_OPTIONS =
+      new HashMap<Integer, ErrorSeverity>() {
+        {
+          put(CICSLexer.XCTL, ErrorSeverity.ERROR);
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java]: This is not one of the operands.

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    if (ctx.getParent().getRuleIndex() == CICSParser.RULE_cics_xctl) {
+      checkReadNextReadPrev((CICSParser.Cics_xctlContext) ctx.getParent());
+    }
+    checkDuplicates(ctx.getParent());
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkReadNextReadPrev(CICSParser.Cics_xctlContext ctx) {
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java]: `checkReadNextReadPrev`? This looks like a copy-paste from previous PRs, please rename.

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  /**
+   * Entrypoint to check CICS XCTL rule options
+   *
+   * @param ctx ParserRuleContext subclass containing options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    if (ctx.getParent().getRuleIndex() == CICSParser.RULE_cics_xctl) {
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java:59]: Given the operands are defined directly in the `cics_xctl` rule, is this check executed operand-count-times?

Comment (sb-broadcom) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSXctlOptionsUtility.java:59]: Not having worked on the code that calls this I can't say for certain, although my guess would be yes. However, I believe the fix for needing to call getParent() is located in the INQUIRE branch, and blocks SUSPEND until it gets merged. Once that is done we should be able to go back and remove the necessity to look at parents in most cases (or create a separate ticket for the same, given that this is used in at least 10 separate options)

**Pull request merged**